File: cpp/171-180/Largest Number.cpp
class Solution {
public:
    string largestNumber(vector<int>& nums) {
        string res;
        bool flag = false;
        vector<string> numStr;
        for (int num : nums) {
            if (num != 0) flag = true;
            numStr.emplace_back(to_string(num));
        }
        auto cmp = [](const string& i, const string& j) {
            return i + j > j + i;
        };
        sort(numStr.begin(), numStr.end(), cmp);
        for (const string& str : numStr) {
            res += str;
        }
        return flag ? res : "0";
    }
};

File: python/171-180/Largest Number.py
class Solution:
    # @param {integer[]} nums
    # @return {string}
    def largestNumber(self, nums):
        num = [str(x) for x in nums]
        num.sort(cmp = lambda x,y : cmp(y+x, x+y))
        return "".join(num).lstrip('0') or '0'